redis-om-spring/src/main/java/com/redis/om/spring/repository/query/RediSearchQuery.java

```diff
@@ -328,15 +342,18 @@ private Object executeQuery(Object[] parameters) {
 
   private Object executeAggregation(Object[] parameters) {
     SearchOperations<String> ops = modulesOperations.opsForSearch(searchIndex);
-    AggregationBuilder aggregation = new AggregationBuilder().load(load);
+    AggregationBuilder aggregation = new AggregationBuilder(value).load(load);
+    for (Map.Entry<String, String> entry : apply.entrySet()) {
+      aggregation.apply(entry.getKey(), entry.getValue());
+    }
     AggregationResult aggregationResult = ops.aggregate(aggregation);
 
     // what to return
     Object result = null;
     if (queryMethod.getReturnedObjectType() == AggregationResult.class) {
       result = aggregationResult;
     } else if (queryMethod.isCollectionQuery()) {
-      result = Collections.EMPTY_LIST;
+      result = Collections.EMPTY_LIST; // TODO: WTF Sam-Bodden?
```

Comment:
At least we should remove the WTF :p

Reply:
I like to "keep it real" :-) I got a PR after this one that addresses this... Aggregations support is pretty light right now.

Reply:
See https://github.com/redis/redis-om-spring/issues/88